app/src/components/file-upload/FileUploadItem.tsx

```diff
@@ -146,6 +151,11 @@ const FileUploadItem: React.FC<IFileUploadItemProps> = (props) => {
     uploadHandler(file, cancelToken, handleFileUploadProgress)
       .then(handleFileUploadSuccess, (error: APIError) => {
         setError(error?.message);
+        setErrorDetails(
+          error?.errors?.map((e) => {
+            return { _id: v4(), message: e?.toString() };
```

Comment:
If this `_id` is being used for providing a React key, using a UUID is generally ill-advised. This is something we need to address in other parts of SIMS so I wouldn't worry too much about it now. The alternative is using the array index, but React also discourages this 🤔

Reply:
If it is just being used as a key, I would rename it to `key` instead of `_id`

Reply:
FWIW, Sonar will at least not complain about code smell by doing this. Kinda curious since depending on the context, the index would actually be more stable than the UUID 🤔